hs-bindgen-libclang/src/HsBindgen/Clang/Core.hs

```diff
@@ -688,17 +692,26 @@ clang_Cursor_isAnonymous cursor =
 -- | Retrieve the integer value of an enum constant declaration as a signed long
 -- long.
 --
--- The @libclang@ docs state:
---
--- > If the cursor does not reference an enum constant declaration, LLONG_MIN is
--- > returned. Since this is also potentially a valid constant value, the kind
--- > of the cursor must be verified before calling this function.
---
--- This is therefore a precondition to calling 'clang_getEnumConstantDeclValue'.
+-- Throws 'CallFailed' if the cursor does not reference an enum constant
+-- declaration.
 --
 -- <https://clang.llvm.org/doxygen/group__CINDEX__TYPES.html#ga6b8585818420e7512feb4c9d209b4f4d>
-clang_getEnumConstantDeclValue :: CXCursor -> IO CLLong
-clang_getEnumConstantDeclValue cursor = do
+clang_getEnumConstantDeclValue ::
+     HasCallStack
+  => CXCursor -- ^ Parent
+  -> CXCursor -> IO CLLong
+clang_getEnumConstantDeclValue parent cursor = do
```

Comment:
I don't like this
- Now the *unchecked* precondition is that `parent` is indeed parent of a `cursor`. 
- The binding is not low-level anymore (not even "low-level with some extra checks").
    - I'd rather experiment with clang_getCursorSemanticParent()

Reply:
I don't mind, if `clang_getCursorSemanticParent` works, also fine with me, but "parents" are not my invention, they are provided explicitly by [`clang_visitChildren`](https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__TRAVERSAL.html#ga5d0a813d937e1a7dcc35f206ad1f7a91), so in that sense it's a natural use of the API that libclang provides.

Reply:
Having parent in `Fold` is ok. But I'd try to keep the actual bindings as close original bindings as possible. `clang_getEnumConstantDeclValue` doesn't take a parent argument.

Reply:
Ok, I will try `clang_getCursorSemanticParent` (but first looking into the segfault in #131).

Reply:
Done in #148.